File: 1.1/src/cz/mg/vulkan/PFNvkGetShaderInfoAMD.java
package cz.mg.vulkan;

public class PFNvkGetShaderInfoAMD extends VkFunctionPointer {
    public PFNvkGetShaderInfoAMD() {
    }

    protected PFNvkGetShaderInfoAMD(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetShaderInfoAMD(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetShaderInfoAMD(long value) {
        setValue(value);
    }

    public PFNvkGetShaderInfoAMD(VkInstance instance) {
        super(instance, new VkString("vkGetShaderInfoAMD"));
    }

    public void call(VkDevice device, VkPipeline pipeline, VkShaderStageFlagBits shaderStage, VkShaderInfoTypeAMD infoType, VkSize pInfoSize, VkObject pInfo, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pipeline != null ? pipeline.getVkAddress() : VkPointer.getNullAddressNative(), shaderStage != null ? shaderStage.getVkAddress() : VkPointer.getNullAddressNative(), infoType != null ? infoType.getVkAddress() : VkPointer.getNullAddressNative(), pInfoSize != null ? pInfoSize.getVkAddress() : VkPointer.NULL, pInfo != null ? pInfo.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void call(long vkaddress, long device, long pipeline, long shaderStage, long infoType, long pInfoSize, long pInfo, long rval);
}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkGetShaderInfoAMD.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkGetShaderInfoAMD_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pipeline, jint shaderStage, jint infoType, jlong pInfoSize, jlong pInfo){
    (void)env;
    (void)jc;
    PFN_vkGetShaderInfoAMD f = (PFN_vkGetShaderInfoAMD)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkPipeline*)jniLongToPointer(pipeline)),
        (VkShaderStageFlagBits)shaderStage,
        (VkShaderInfoTypeAMD)infoType,
        ((size_t*)jniLongToPointer(pInfoSize)),
        ((void*)jniLongToPointer(pInfo))
    );
}

